### ISSUE TITLE
workflow not work

### ISSUE THREAD

**Issue by kakaluote** [open-mmlab/mmdetection#271]
when i config workflow = [('train', 2),('val',1)], train crash by this
`assert len(data_loaders) == len(workflow)`

because  len(data_loaders) always 1

**Comment by hellock** [comment]
If you specify 2 phases in the workflow, you need 2 dataloaders. In detection tasks we implement mAP evaluation during training to serve as validation, so the val phase is unnecessary. If you want to run val epochs and see the validation loss, you can modify the codes to build 2 dataloaders for the runner.

**Comment by alexwq100** [comment]
hi ,@hellock,
you say " In detection tasks we implement mAP evaluation during training to serve as validation, so the val phase is unnecessary." where is it,in the tools/train.py I haven't found.

**Comment by tdiekel** [comment]
@alexwq100 [Look here](https://github.com/open-mmlab/mmdetection/issues/171#issuecomment-446434197)
